### PR TITLE
Nisp 1524 - Fix for screen reader

### PR DESCRIPTION
**Pull request by roundrobin99**

### CODE REVIEW

File: app/uk/gov/hmrc/nisp/views/includes/nirecordtaxyear.scala.html [hunk 1]
```diff
@@ -28,12 +28,12 @@
...
     }
         <div class="ni-wrapper">
-            <div class="ni-years"> @niRecordTaxYear.displayableTaxYear</div>
+            <div class="ni-years"><abbr title="@niRecordTaxYear.taxYear">@niRecordTaxYear.taxYear</abbr> - <abbr title="@{{niRecordTaxYear.taxYear+1}.toString.substring(2,4)}">@{{niRecordTaxYear.taxYear+1}.toString.substring(2,4)}</abbr></div>
             @if(niRecordTaxYear.qualifying){
                 <div class="inactive">@Messages("nisp.nirecord.fullyear")</div>
             } else {
                 <div class="ni-notfull">@Messages("nisp.nirecord.gap")</div>
-                <div class="ni-action"><a href="#">@Html(Messages("nisp.nirecord.gap.viewdetails", niRecordTaxYear.displayableTaxYear))</a></div>
+                <div class="ni-action"><a href="#">@Html(Messages("nisp.nirecord.gap.viewdetails", niRecordTaxYear.taxYear.toString))</a></div>
```
Comment (andrewgee) [inline app/uk/gov/hmrc/nisp/views/includes/nirecordtaxyear.scala.html]: This won't read the same as what was expected, will it? It only has the first part of the tax year?

File: conf/messages [hunk 2]
```diff
@@ -245,7 +245,7 @@ nisp.nirecord.gapsinyourrecord.line1= in paid employment and had low earnings
 nisp.nirecord.gapsinyourrecord.line2= unemployed and not claiming benefit
 nisp.nirecord.gapsinyourrecord.line3= self-employed but did not pay contributions because of small profits
 nisp.nirecord.gapsinyourrecord.line4= living abroad
-nisp.nirecord.gap.viewdetails = View <span class="visuallyhidden">{0} </span>details
+nisp.nirecord.gap.viewdetails = View <span class="visuallyhidden"><abbr title="{0}">{0}</abbr></span>details
```
Comment (andrewgee) [inline conf/messages]: What does the abbr tag achieve here?

File: public/javascript/nirecord.js [hunk 3]
```diff
@@ -3,7 +3,7 @@ $(document).ready(function(){
   var allHeaders = $('.accordion > dt');
 
   $('.accordion > dt.expandable').click(function() {
-    var taxYear = $(this).find(".ni-years").text()
+    var taxYear = $(this).find(".ni-years > abbr:first-child").text()
```
Comment (andrewgee) [inline public/javascript/nirecord.js]: You lose the abbr tag that you created in this JavaScript.

File: app/uk/gov/hmrc/nisp/views/includes/nirecordtaxyear.scala.html (hunk 1 above)
Comment (roundrobin99) [inline app/uk/gov/hmrc/nisp/views/includes/nirecordtaxyear.scala.html]: Yes.There is no other way I could find.

Comment (andrewgee) [inline app/uk/gov/hmrc/nisp/views/includes/nirecordtaxyear.scala.html]: Why is it different to line 31?

File: conf/messages (hunk 2 above)
Comment (roundrobin99) [inline conf/messages]: It helps with the date being understood by the screen reader.

File: public/javascript/nirecord.js (hunk 3 above)
Comment (roundrobin99) [inline public/javascript/nirecord.js]: Good point.

File: conf/messages (hunk 2 above)
Comment (andrewgee) [inline conf/messages]: But it's not an abbreviation and it's got the same text inside the attribute and the innerHtml of that tag.

File: app/uk/gov/hmrc/nisp/views/includes/nirecordtaxyear.scala.html (hunk 1 above)
Comment (roundrobin99) [inline app/uk/gov/hmrc/nisp/views/includes/nirecordtaxyear.scala.html]: That's the only way the year can be read by the screen reader

File: conf/messages (hunk 2 above)
Comment (roundrobin99) [inline conf/messages]: It doesn't matter. It is s trick for reading dates properly.

File: public/javascript/nirecord.js (hunk 3 above)
Comment (roundrobin99) [inline public/javascript/nirecord.js]: I have added the abbr tag now in the javascript.

File: conf/messages (hunk 2 above)
Comment (roundrobin99) [inline conf/messages]: I have removed the "title" attribute  as it is only needed if you have a full date.

File: app/uk/gov/hmrc/nisp/views/includes/nirecordtaxyear.scala.html (hunk 1 above)
Comment (andrewgee) [inline app/uk/gov/hmrc/nisp/views/includes/nirecordtaxyear.scala.html]: Where did you find this out?

Comment (roundrobin99) [inline app/uk/gov/hmrc/nisp/views/includes/nirecordtaxyear.scala.html]: http://lab.dotjay.co.uk/tests/screen-readers/date-time/

Comment (andrewgee) [inline app/uk/gov/hmrc/nisp/views/includes/nirecordtaxyear.scala.html]: I can't see where it mentions the abbr tag?

Comment (roundrobin99) [inline app/uk/gov/hmrc/nisp/views/includes/nirecordtaxyear.scala.html]: Try some of the links

Comment (andrewgee) [inline app/uk/gov/hmrc/nisp/views/includes/nirecordtaxyear.scala.html]: This page? http://lab.dotjay.co.uk/tests/screen-readers/microformats/datetime-design-pattern/

It explains that abbr isn't the correct use and causes accessibility issues?

Comment (roundrobin99) [inline app/uk/gov/hmrc/nisp/views/includes/nirecordtaxyear.scala.html]: Yeah, but it seems to be the only one that works. The Time microformat is only supported by recent versions of some screen readers, so the abbr is the only one that works.

Comment (dspork) [inline app/uk/gov/hmrc/nisp/views/includes/nirecordtaxyear.scala.html]: Should we ask our accessibility expert?

Comment (roundrobin99) [inline app/uk/gov/hmrc/nisp/views/includes/nirecordtaxyear.scala.html]: Sure. Although I don't think he can tell you what HTML to use

Comment (roundrobin99) [inline app/uk/gov/hmrc/nisp/views/includes/nirecordtaxyear.scala.html]: I have replaced the ABBR tag with the HTML5 TIME tag which seems to work.

**Comment by dspork** [comment]
This seems more suitable than an <abbr> tag, @swatcats143 can you test this, and then I will merge.

**Comment by swatcats143** [comment]
@dspork not working as expected so escalted back to @roundrobin99 I will put the ticket in progress

**Comment by dspork** [comment]
Closing this as we are going to review the user need with the accessibility team.

**Pull request closed without merging**